src/waveform/renderers/waveformmarkset.cpp

```diff
@@ -30,16 +29,15 @@ void WaveformMarkSet::setup(const QString& group, const QDomNode& node,
     QDomNode child = node.firstChild();
     while (!child.isNull()) {
         if (child.nodeName() == "DefaultMark") {
-            m_defaultMark.setup(group, child, context, signalColors);
+            m_pdefaultMark = std::make_unique<WaveformMark>(group, child, context, signalColors);
             hasDefaultMark = true;
         } else if (child.nodeName() == "Mark") {
-            WaveformMarkPointer pMark(new WaveformMark());
-            pMark->setup(group, child, context, signalColors);
+            WaveformMarkPointer pMark(new WaveformMark(group, child, context, signalColors));
 
             bool uniqueMark = true;
-            if (pMark->m_pPointCos) {
+            if (!pMark->is_m_pPointCos_Null()) {
```

Comment:
One of the main OO principles: Encapsulation. This is already violated by the name of the function.